spec/support/shared_examples_policy_machine.rb

```diff
@@ -1135,14 +1134,116 @@
+
+    before do
+
+      [one_fish, two_fish, red_one].each do |object|
+        policy_machine.add_association(ua, writer, object)
+      end
+
+      policy_machine.add_assignment(reader, read)
+      policy_machine.add_assignment(reader, write)
+      policy_machine.add_assignment(reader, edit)
+      policy_machine.add_assignment(reader, prohib_write)
+      policy_machine.add_assignment(reader, prohib_edit)
+      policy_machine.add_assignment(writer, read)
+      policy_machine.add_assignment(writer, write)
+      policy_machine.add_assignment(writer, edit)
+      policy_machine.add_assignment(writer, prohib_edit)
 
+      policy_machine.add_assignment(user, ua)
+      policy_machine.add_assignment(red_one, oa)
+      policy_machine.add_assignment(blue_one, oa)
+
+      policy_machine.add_association(ua, reader, oa)
     end
 
-  end
+    context "when no options are provided" do
+      it 'returns all privileges the user has on the specified object' do
+        expected_privileges = [
+          [user, read, one_fish],
+          [user, write, one_fish]
+        ]
 
+        expect(policy_machine.scoped_privileges(user, one_fish)).to match_array(expected_privileges)
+      end
 
-  describe 'batch_find' do
+      it 'returns all privileges the user has on the specified object attribute' do
+        expected_privileges = [
+          [user, read, oa]
+        ]
+
+        expect(policy_machine.scoped_privileges(user, oa)).to match_array(expected_privileges)
+      end
+
+      it 'does not returned prohibited privileges' do
+        unexpected_privilege = [user, edit, one_fish]
+        expect(policy_machine.scoped_privileges(user, one_fish)).not_to include(unexpected_privilege)
+      end
+    end
 
+    context "when the 'filters' option is provided" do
+      let(:options) { { user_attributes: { unique_identifier: writer.unique_identifier } } }
+
+      it 'returns all privileges the user has on the specified object with filters applied' do
+        expected_privileges = [
+          [user, read, one_fish],
+          [user, write, one_fish]
+        ]
+
+        expect(policy_machine.scoped_privileges(user, one_fish, options)).to match_array(expected_privileges)
+      end
+    end
+
+    context "when the 'ignore_prohibitions' option is provided" do
+      let(:options) { { ignore_prohibitions: true } }
+
+      it 'returns all privileges the user has on the specified object ignoring prohibitions' do
+        expected_privileges = [
+          [user, read, one_fish],
+          [user, write, one_fish],
+          [user, edit, one_fish]
+        ]
+
+        expect(policy_machine.scoped_privileges(user, one_fish, options)).to match_array(expected_privileges)
+      end
+    end
+
+    context "when the 'include_prohibitions' option is provided" do
+      let(:options) { { include_prohibitions: true } }
+
+      it 'returns all privileges and prohibitions the user has on the specified object' do
```

Comment:
mind adding a spec for behavior when a user is prohibited on an object but doesn't have privilege on it? say:
```
        parent (prohibition)
  child 1 (privilege)   child 2 (no direct assignment)
```